app/src/main/java/com/berniesanders/fieldthebern/controllers/PhotoController.java

```diff
@@ -112,15 +125,36 @@ public void onResult(int requestCode, int resultCode, Intent data) {
         if (resultCode == android.app.Activity.RESULT_OK) {
           Uri imageUri = data.getData();
 
-          //TODO this code sucks
-          // How is it the api for interacting with the media store
-          // is still manually iterating with a cursor...?!
-          long id = Long.parseLong(imageUri.getLastPathSegment()
-              .substring(imageUri.getLastPathSegment().lastIndexOf(":") + 1));
+          String imagePath = imageUri.getPath();
+          //If the uri comes from the default document picker, we don't have to worry about the
+          // extension. Android will already have made sure it is an image.
+          if (!"com.android.providers.media.documents".equals(imageUri.getAuthority())) {
```

Comment:
When I get the chance I need to check the minSdkVersion for the app. Previous to KitKat, this was "media/external/images/media"